src/main/java/org/apache/sling/api/uri/SlingUriBuilder.java

```diff
@@ -227,8 +227,8 @@ public static SlingUriBuilder createFrom(@NotNull URI uri, @Nullable ResourceRes
         String path = uri.getRawPath();
         boolean pathExists = isNotBlank(path);
         String uriQuery = uri.getRawQuery();
-        boolean schemeSpecificRelevant = !pathExists && uriQuery == null;
         String uriHost = uri.getHost();
+        boolean schemeSpecificRelevant = !pathExists && uriQuery == null && uriHost == null;
```

Comment:
Shouldn't we rather use https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#isOpaque--? Or are there any other non-opaque URIs which ever return true?